src/pages/iou/IOUCurrencySelection.js

```diff
@@ -105,75 +84,56 @@ class IOUCurrencySelection extends Component {
-        const iouType = lodashGet(this.props.route, 'params.iouType', CONST.IOU.MONEY_REQUEST_TYPE.REQUEST);
-        const reportID = lodashGet(this.props.route, 'params.reportID', '');
-        return (
-            <ScreenWrapper includeSafeAreaPaddingBottom={false}>
-                {({safeAreaPaddingBottomStyle}) => (
-                    <>
-                        <HeaderWithBackButton
-                            title={this.props.translate('iOUCurrencySelection.selectCurrency')}
-                            onBackButtonPress={() => Navigation.goBack(ROUTES.getMoneyRequestRoute(iouType, reportID))}
-                        />
-                        <OptionsSelector
-                            sections={this.getSections()}
-                            onSelectRow={this.confirmCurrencySelection}
-                            value={this.state.searchValue}
-                            onChangeText={this.changeSearchValue}
-                            textInputLabel={this.props.translate('common.search')}
-                            headerMessage={headerMessage}
-                            safeAreaPaddingBottomStyle={safeAreaPaddingBottomStyle}
-                            initiallyFocusedOptionKey={_.get(
-                                _.find(this.state.currencyData, (currency) => currency.currencyCode === this.getSelectedCurrencyCode()),
-                                'keyForList',
-                            )}
-                            shouldHaveOptionSeparator
-                        />
-                    </>
-                )}
-            </ScreenWrapper>
-        );
-    }
+        const isEmpty = searchValue.trim() && !filteredCurrencies.length;
+
+        return {
+            initiallyFocusedOptionKey: _.get(
+                _.find(filteredCurrencies, (currency) => currency.currencyCode === selectedCurrencyCode),
+                'keyForList',
+            ),
+            sections: isEmpty
+                ? []
+                : [
+                      {
+                          title: translate('iOUCurrencySelection.allCurrencies'),
+                          data: filteredCurrencies,
+                          shouldShow: true,
+                          indexOffset: 0,
+                      },
+                  ],
+            headerMessage: isEmpty ? translate('common.noResultsFound') : '',
+        };
+    }, [currencyList, searchValue, selectedCurrencyCode, translate]);
+
+    return (
+        <ScreenWrapper includeSafeAreaPaddingBottom={false}>
+            {({safeAreaPaddingBottomStyle}) => (
+                <>
+                    <HeaderWithBackButton
+                        title={translate('iOUCurrencySelection.selectCurrency')}
+                        onBackButtonPress={() => Navigation.goBack(ROUTES.getIouRequestRoute(Navigation.getTopmostReportId()))}
```

Comment:
`ROUTES.getIouRequestRoute()` doesn't currently exist which is causing the regression I commented on, will see if I can figure out what this is supposed to be and add it.

Reply:
Yep! That's what I ended up going with - https://github.com/Expensify/App/pull/22135/files#diff-496700da68574808c5981ac4884e39d73ea02df4aa59bd5869e0c9227defe4c2R119 - it's weird, I looked through every commit in this PR and it never actually changed away from that in any of them. But it ends up being different in the end. I'm not sure how that happened

Reply:
ahh that makes sense.

Reply:
Thanks both, Yeah my first guess was the merge as I recall some conflicts. I must have missed one when resolving.